lib/server/services/detector/getOrCreateAllPhysicalDetectorsByName.js

```diff
@@ -16,17 +16,25 @@ const { createDetector } = require('./createDetector.js');
 const { getDetector } = require('./getDetector.js');
 
 /**
- * Find all the detectors corresponding to a list of detectors names. If any of the name do not correspond to a detector, creates it
+ * Find all the physical detectors corresponding to a list of detectors names. If any of the name do not correspond to a detector, creates it
  *
  * @param {string[]} [detectorNames] the list of detectors name
  * @return {Promise<SequelizeDetector[]>} the list of detectors
  */
-exports.getOrCreateAllDetectorsByName = async (detectorNames) => {
+exports.getOrCreateAllPhysicalDetectorsByName = async (detectorNames) => {
     if (detectorNames.length === 0) {
         return [];
     }
 
     const detectors = await getDetectorsByNames(detectorNames);
+    const nonPhyscial = detectors.filter(({ isPhysical }) => !isPhysical);
+
+    if (nonPhyscial.length === 1) {
+        throw new Error(`Detector ${nonPhyscial[0].name} exists, but it is not physical`);
```

Comment:
As always, code simplicity is the key. It's nicer to have valid English sentences, but that's too much code added for really not much. `Detectors [MCH] exist, but are not physical` is good enough for errors reporting.

However, I think that we should not throw in that case but make the detectors physical to be consistent.

Reply:
applied